executable-spec/src/Cardano/Ledger/Spec/STS/Update/Ideation.hs

```diff
@@ -136,14 +139,16 @@ instance HashAlgorithm hashAlgo => STS (IDEATION hashAlgo) where
           , st@St { subsips
                   , wssips
                   , wrsips
+                  , sipdb
                   , ballots
                   }
           , sig
           ) <- judgmentContext
       case sig of
         Submit sipc sip -> do
           Data.author sip ∈ dom participants ?! InvalidAuthor (Data.author sip)
-          sip ∉ subsips ?! SIPAlreadySubmitted sip
+          sip ∉ range subsips ?! SIPAlreadySubmitted sip
+          sip ∉ range sipdb ?! SIPSubmittedAlreadyRevealed sip
```

Comment:
We can remove the check `sip ∉ range sipdb ?! SIPSubmittedAlreadyRevealed sip`
and replace `sip ∉ range subsips ?! SIPAlreadySubmitted sip`
with 
`sipc ∉ dom subsips ?! SIPAlreadySubmitted sip`
But we should also stop deleting `subsips` for the check to be safe

Reply:
I have replaced `sip ∉ range subsips ?! SIPAlreadySubmitted sip`
with `(Data.commit sipc) ∉ dom subsips ?! SIPAlreadySubmitted sip`, in order not to depend on `SIP` for the check.
Also I have removed the `sip ∉ range sipdb ?! SIPSubmittedAlreadyRevealed sip`
altogether and stopped emptying `subsips`, so there is no need to check into `sipdb` for an already revealed SIP at commit time, because all revealed SIPs are also included in `subsips`